week-1/3-extra/3-magic-8-ball.js

```diff
@@ -42,20 +42,66 @@ My sources say no.
 Outlook not so good.
 Very doubtful.
 */
-
+let veryPositive = [
+    "It is certain.",
+    "It is decidedly so.",
+    "Without a doubt.",
+    "Yes - definitely.",
+    "You may rely on it."
+]
+let positive = [
+    "As I see it, yes.",
+    "Most likely.",
+    "Outlook good.",
+    "Yes.",
+    "Signs point to yes."
+]
+let negative = [ 
+              "Reply hazy, try again.",
+              "Ask again later.",
+              "Better not tell you now.",
+              "Cannot predict now.",
+              "Concentrate and ask again."
+]
+let veryNegative = [
+              "Don't count on it", 
+              "My reply is no", 
+              "My sources say no.", 
+              "Outlook not so good.", 
+              "Very doubtful."
+]
+let posibleTypes = [
+              "very positive",
+              "positive",
+              "negative",
+              "very negative"
+]
+let possibleAnswers = [ veryPositive, positive, negative, veryNegative  ]
 // This should log "The ball has shaken!"
 // and return the answer.
-function shakeBall() {}
+function shakeBall() {
+  console.log("The ball has shaken!")
+  let answerType = Math.floor(Math.random() * 4)
+  let selectedAnswer = possibleAnswers[ answerType ]
+  
+  return selectedAnswer[Math.floor(Math.random() * 5) ]
+}
 
 // The answer should come from shaking the ball
-let answer;
+//let answer = shakeBall();
 
 // When checking the answer, we should tell someone if the answer is
 // - very positive
 // - positive
 // - negative
 // - very negative
-function checkAnswer() {}
+function checkAnswer(answer) {
+  for (let answerType = 0; answerType < possibleAnswers.length; ++answerType ){
+    if (possibleAnswers[ answerType ].includes(answer) )
```

Comment:
So this will work but you're double checking

lets say `answerType === 2`
then `possibleAnswers[ answerType ] => "negative"
then `"negative".includes("negative")` will always be true

`.includes` can be used on both `strings` and `arrays`

just use `possibleAnswers.includes(answer)`

also should this be `posibleTypes`?

That way we don't need to track the array index and can express things a lot simpler.
See below (for fun) I don't expect you to be jumping straight into arrow functions but you might be interested :)

Reply:
Ah....`possibleAnswers[ 2 ] => [ 
              "Reply hazy, try again.",
              "Ask again later.",
              "Better not tell you now.",
              "Cannot predict now.",
              "Concentrate and ask again."
]`

My code uses 2 arrays, `possibleAnswers` is an array of arrays defined on line 79. So, `possibleAnswers[ answerType ] => [ 
              "Reply hazy, try again.",
              "Ask again later.",
              "Better not tell you now.",
              "Cannot predict now.",
              "Concentrate and ask again."
]`

So, those group of answers are part of the 'negative' category, hence `answerType === 2`
And I hold the answer categories in `posibleTypes = [
              "very positive",
              "positive",
              "negative",
              "very negative"
]`

Hmmmm...just noticed that I misspelt possible...ah well!